File: Data Structures & Algorithms/top-k-elements-in-list/submission-4.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        counter = {}

        for num in nums:
            counter[num] = counter.get(num, 0) + 1

        reversed_list = []

        for key, value in counter.items():
            reversed_list.append([value, key])

        reversed_list.sort()

        answer = []

        while len(answer) < k:
            answer.append(reversed_list.pop()[1])

        return answer

        

File: Data Structures & Algorithms/majority-element-ii/submission-4.cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
        vector<int> res;
        int totalCount = nums.size();
        unordered_map<int,int> counter;

        for (int num : nums) {
            counter[num] += 1;
        }

        for (auto [num,freq] : counter ) {
            if (freq > totalCount / 3) {
                res.push_back(num);
            }
        }

        return res;
    }
};